core/services/editor/BlockAssetRegister.php

```diff
@@ -0,0 +1,219 @@
+<?php
+
+namespace EventEspresso\core\services\editor;
+
+use EventEspresso\core\domain\DomainInterface;
+use EventEspresso\core\exceptions\InvalidDataTypeException;
+use EventEspresso\core\services\assets\AssetRegisterInterface;
+use EventEspresso\core\services\assets\Registry;
+
+defined('EVENT_ESPRESSO_VERSION') || exit;
+
+
+
+/**
+ * Class BlockAssetRegister
+ * Abstract parent for classes that handle asset registration for one or more editor blocks that share the same assets
+ *
+ * @package EventEspresso\core\services\editor
+ * @author  Brent Christensen
+ * @since   $VID:$
+ */
+abstract class BlockAssetRegister implements AssetRegisterInterface
+{
+
+    /**
+     * @var DomainInterface $domain
+     */
+    private $domain;
+
+    /**
+     * @var Registry $registry
+     */
+    private $registry;
+
+    /**
+     * @var string $script_handle
+     */
+    private $script_handle;
+
+    /**
+     * @var array $script_dependencies
+     */
+    private $script_dependencies;
+
+    /**
+     * @var string $style_handle
+     */
+    private $style_handle;
+
+    /**
+     * @var array $style_dependencies
+     */
+    private $style_dependencies;
+
+
+    /**
+     * BlockAssetRegister constructor.
+     *
+     * @param string          $script_handle
+     * @param array           $script_dependencies
+     * @param string          $style_handle
+     * @param array           $style_dependencies
+     * @param DomainInterface $domain
+     * @param Registry        $registry
+     * @throws InvalidDataTypeException
+     */
+    public function __construct(
+        $script_handle,
+        array $script_dependencies,
+        $style_handle,
+        array $style_dependencies,
```

Comment:
based on my previous comments about the difference between `editor_script` and `script` (and etc with styles).  I wonder if this interface needs tweaking.  Maybe a VO or something to allow for multiple script handles and their dependencies to be set on this class?